Slide for Reddit/AutoCache.swift

```
@@ -4,197 +4,269 @@
+                        if self.cancel {
+                            return
+                        }
+                        let newLinks = listing.children.compactMap({ $0 as? Link })
+                        var converted: [SubmissionObject] = []
+                        for link in newLinks {
+                            let newRS = SubmissionObject.linkToSubmissionObject(submission: link)
+                            converted.append(newRS)
+                        }
+                        let values = PostFilter.filter(converted, previous: [], baseSubreddit: sub).map { $0 as! SubmissionObject }
+                        
+                        self.saveSubmissionsCoreData(subreddit: sub, links: values)
+                        self.preloadImages(values)
+                        self.cacheComments(index, commentIndex: 0, currentLinks: values, done: 0, failed: 0, progress: progress, completion: completion)
+                    }
+                })
+            } catch {
+                print(error)
+            }
+        }
+    }
+    
+    var toBeExecuted: [() -> Void] = []
+
+    func saveSubmissionsCoreData(subreddit: String, links: [SubmissionObject]) {
+        let context = SlideCoreData.sharedInstance.backgroundContext
+        context.performAndWait {
+            var ids = [String]()
+            for link in links {
+                ids.append(link.getId())
+                
+                _ = link.insertSelf(into: context, andSave: false)
+            }
+            
+            var subredditPosts: SubredditPosts! = nil
+            let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: "SubredditPosts")
+            let predicate = NSPredicate(format: "subreddit = %@", subreddit)
+            fetchRequest.predicate = predicate
+            do {
+                let results = try context.fetch(fetchRequest) as! [SubredditPosts]
+                subredditPosts = results.first
+            } catch {
+                
```

Log the error. No empty catch blocks!